darts/dataprocessing/utils.py

```diff
@@ -0,0 +1,149 @@
+
+            super().__init__(name=name, validators=None)  # TODO: Add possibility to specify validators ?
+
+            self._transform = transform
+            self._inverse_transform = inverse_transform
+            self._fit = fit
+
+            self._name = name
+
+        def transform(self, data: T, *args, **kwargs) -> T:
+            return self._transform(data, *args, **kwargs)
+
+        if InvertibleDataTransformer in parent_classes:
+            def inverse_transform(self, data: T, *args, **kwargs) -> T:
+                return self._inverse_transform(data, *args, **kwargs)
+
+        if FittableDataTransformer in parent_classes:
+            def fit(self, data: T) -> 'Type[BaseDataTransformer[T]]':
+                self._fit(data)
+                return self
+
+    return _WrapperDataTransformer(transform, inverse_transform, fit, name)
+
+
+def _get_parent_classes(inverse_transform: Optional[_callable_with_args_kwargs] = None,
+                        fit: Optional[_callable_with_args_kwargs] = None):
+    parent_classes = []
+    if inverse_transform is not None:
+        parent_classes.append(InvertibleDataTransformer)
+    if fit is not None:
+        parent_classes.append(FittableDataTransformer)
+
+    if not parent_classes:
+        parent_classes.append(BaseDataTransformer)
+
+    return parent_classes
+
+
+def data_transformer_from_ts_functions(transform: _callable_with_args_kwargs[TimeSeries],
+                                       inverse_transform: Optional[_callable_with_args_kwargs[TimeSeries]] = None,
+                                       fit: Optional[Callable[[TimeSeries], Any]] = None,
+                                       name: str = "FromTSWrappedDataTransformer") -> Type[BaseDataTransformer[TimeSeries]]:  # noqa: E501
+    """
+    Utility function to create a data transformer from functions taking as input TimeSeries.
+    All functions except transform are optional.
+
+    Parameters
+    ----------
+    transform
+        Function taking TimeSeries as an input and returning TimeSeries.
+    inverse_transform
+        Function taking TimeSeries as an input and returning TimeSeries.
+    fit
+        Function taking TimeSeries as an input.
+    name
+        Data transformer's name will be set to this.
+
+    Returns
+    -------
+    Type[BaseDataTransformer[TimeSeries]]
+        Data transformer created from the provided functions.
+    """
+    return _create_wrapper_data_transformer(_get_parent_classes(inverse_transform, fit),
+                                            transform,
+                                            inverse_transform,
+                                            fit,
+                                            name)
+
+
+def data_transformer_from_values_functions(transform: _callable_with_args_kwargs[np.ndarray],
+                                           inverse_transform: Optional[_callable_with_args_kwargs[np.ndarray]] = None,
```

Comment:
I'm not sure yet whether we really need both "versions" of the same function (namely `data_transformer_from_ts_functions()` and `data_transformer_from_values_functions()`, nor do I really like the functions' names. I might change my mind implementing common transformers, but leaving this comment here for the record.

Reply:
What do you think about calling it a `Transformation` ?

Reply:
or maybe keeping `transformer` in the name makes actually more sense I dont know like in sklearn you have `FunctionTranformer`: https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.FunctionTransformer.html#sklearn.preprocessing.FunctionTransformer